File: user-resource-service/src/main/java/com/movies/user/util/SecurityUtils.java
package com.movies.user.util;

import com.movies.common.AuthorizedUser;
import com.movies.common.user.UserTo;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

import static java.util.Objects.requireNonNull;

public class SecurityUtils {
    private static AuthorizedUser safeGet() {
        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
        if (auth == null)
            return null;
        Object principal = auth.getPrincipal();
        return (principal instanceof AuthorizedUser) ? (AuthorizedUser) principal : null;
    }

    public static AuthorizedUser getAuthorizedUser() {
        AuthorizedUser user = safeGet();
        requireNonNull(user, "No authorized user found");
        return user;
    }

    public static UserTo authUser() {
        return getAuthorizedUser().getUserTo();
    }

    public static int authUserId() {
        return getAuthorizedUser().getId();
    }
}

File: movies-list-service/src/main/kotlin/com/movies/list/utils/SecurityUtils.kt
package com.movies.list.utils

import com.movies.common.AuthorizedUser
import com.movies.common.user.UserTo
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.common.exceptions.UnauthorizedUserException

object SecurityUtils {
    private fun safeGet(): AuthorizedUser? {
        val auth = SecurityContextHolder.getContext().authentication
                ?: return null
        val principal = auth.principal
        return if (principal is AuthorizedUser) principal else null
    }

    fun getAuthorizedUser(): AuthorizedUser =
        safeGet() ?: throw UnauthorizedUserException("No authorized user found")

    fun authUser(): UserTo = getAuthorizedUser().userTo

    fun authUserId(): Int = getAuthorizedUser().id

    fun authUserIdOrAnonymous() = safeGet()?.id?.toString() ?: "[anonymous]"
}